Elasticsearch Platform Service Docs. (#3542)
mesosphere/dcos-docs-site: pages/dkp/kommander/1.3/projects/platform-services/platform-services-catalog/elasticsearch/index.md

---
layout: layout.pug
navigationTitle: Elasticsearch on Kommander
render: mustache
title: Elasticsearch on Kommander
menuWeight: 3
excerpt: Elasticsearch Day 2 Operations
beta: false
---

### Elasticsearch overview

[Elasticsearch](https://www.elastic.co/what-is/elasticsearch) is a distributed, RESTful search and analytics engine capable of addressing a growing number of use cases. As the core of the Elastic Stack, Elasticsearch centrally stores your data for lightning fast search, fine‑tuned relevancy, and powerful analytics that scale with ease.

### Kommander catalog

Kommander catalog adds integration for Elasticsearch in [Helm-based drivers](https://docs.d2iq.com/dkp/kommander/1.3/projects/platform-services/helm-based/)

To access the catalog:
#include /dkp/kommander/1.3/include/kommander-catalog-drilldown.tmpl

### Install

From the [Project Catalog](/dkp/kommander/1.3/projects/platform-services/) select the desired version of Elasticsearch, and select Deploy.

> Below is an example of what should appear in the Kommander UI. The dialog is populated with appropriate defaults:

Reviewer: ```suggestion
The Kommander UI should resemble the following image. The dialog is populated with appropriate defaults:
```

The following services are exposed by Elasticsearch:
```
$ kubectl get services
NAME                                TYPE        CLUSTER-IP    EXTERNAL-IP   PORT(S)             AGE
elasticsearch-oss-client            ClusterIP   10.0.30.140   <none>        9200/TCP,9300/TCP   91m
elasticsearch-oss-client-headless   ClusterIP   None          <none>        9200/TCP,9300/TCP   91m
elasticsearch-oss-data              ClusterIP   10.0.5.4      <none>        9200/TCP,9300/TCP   91m
elasticsearch-oss-data-headless     ClusterIP   None          <none>        9200/TCP,9300/TCP   91m
elasticsearch-oss-ingest            ClusterIP   10.0.29.173   <none>        9200/TCP,9300/TCP   91m
elasticsearch-oss-ingest-headless   ClusterIP   None          <none>        9200/TCP,9300/TCP   91m
elasticsearch-oss-master            ClusterIP   10.0.48.163   <none>        9200/TCP,9300/TCP   91m
elasticsearch-oss-master-headless   ClusterIP   None          <none>        9200/TCP,9300/TCP   91m
```

Elasticsearch provides comprehensive [REST API](https://www.elastic.co/guide/en/elasticsearch/reference/current/rest-apis.html)

> The following example shows how to query Elasticsearch REST API:

Reviewer: ```suggestion
The following example shows how to query the Elasticsearch REST API:
```

    ```
    $ kubectl get pods
    NAME                         READY   STATUS    RESTARTS   AGE
    elasticsearch-oss-client-0   1/1     Running   0          70m
    elasticsearch-oss-data-0     1/1     Running   0          70m
    elasticsearch-oss-data-1     1/1     Running   0          70m
    elasticsearch-oss-data-2     1/1     Running   0          70m
    elasticsearch-oss-data-3     1/1     Running   0          2m8s
    elasticsearch-oss-master-0   1/1     Running   0          70m
    elasticsearch-oss-master-1   1/1     Running   0          70m
    elasticsearch-oss-master-2   1/1     Running   0          70m
    ```
### Monitoring

Kommander includes Prometheus and Grafana as part of the federated [Workspace Platform Services](/dkp/kommander/1.3/workspaces/workspace-platform-services) along with [Centralized Monitoring](/dkp/kommander/1.3/centralized-monitoring/).

The Kommander Catalog includes [Elasticsearch-Exporter](/dkp/kommander/1.3/projects/platform-services/platform-services-catalog/elasticsearch-exporter/) to be used in conjunction with Elasticsearch to export metrics to Prometheus and Grafana.

### External Access

> The Kommander Catalog includes [Kibana](/dkp/kommander/1.3/projects/platform-services/platform-services-catalog/kibana/) which provides a graphical way to view data stored in Elasticsearch

Reviewer 2: Is the heading `External Access` appropriate here? Because there is no external access shown via NodePort or LoadBalancer.
